### PR TITLE
Replace deprecated PHPStan\Type\Constant\ConstantArrayType::findTypeAndMethodName() call

### LINKED ISSUE #484: Call to deprecated method findTypeAndMethodName() of class PHPStan\Type\Constant\ConstantArrayType
Builds are failing (like in #476) with this issue.

### PR DESCRIPTION
**Pull request by mxr576**
Closes #484

### CODE REVIEW

**Comment by mxr576** [comment]
Maybe it is time to drop PHPStan 1.6 support? :grin:

**Comment by mxr576** [comment]
I guess fixing the PHP 8.0 issues should not be the scope of this PR... so ready for review.

File: src/Rules/Drupal/RenderCallbackRule.php [hunk 1]
```diff
@@ -162,15 +162,17 @@ private function doProcessNode(Node\Expr $node, Scope $scope, string $keyChecked
...
+            $typeAndMethodNames = $type->findTypeAndMethodNames();
+            if ($typeAndMethodNames === []) {
                 throw new \PHPStan\ShouldNotHappenException();
             }
 
-            if (!$trustedCallbackType->isSuperTypeOf($typeAndMethodName->getType())->yes()) {
-                return RuleErrorBuilder::message(
-                    sprintf("%s callback class '%s' at key '%s' does not implement Drupal\Core\Security\TrustedCallbackInterface.", $keyChecked, $typeAndMethodName->getType()->describe(VerbosityLevel::value()), $pos)
-                )->line($errorLine)->tip('Change record: https://www.drupal.org/node/2966725.')->build();
+            foreach ($typeAndMethodNames as $typeAndMethodName) {
```
Comment (mxr576) [inline src/Rules/Drupal/RenderCallbackRule.php:170]: I must admit, this was a no brainer approach :P

**Comment by mglaman** [comment]
Trying to reduce the non-related errors, via https://github.com/mglaman/phpstan-drupal/pull/487

File: src/Rules/Drupal/RenderCallbackRule.php [hunk 2]
```diff
@@ -219,7 +221,7 @@ private function getType(Node\Expr $node, Scope $scope):  Type
                 }
             }
         } elseif ($type instanceof ConstantStringType) {
-            if ($type->isClassString()) {
+            if (!$type->isClassStringType()->no()) {
```
Comment (mglaman) [inline src/Rules/Drupal/RenderCallbackRule.php:224]: Asked about this here: https://github.com/phpstan/phpstan/discussions/8338

**Comment by mglaman** [comment]
> Maybe it is time to drop PHPStan 1.6+1.7 support? 😁

Yeah, if Drupal core is on 1.8.x, I'm all for dropping these, then. I'll do it in another PR.

**Comment by mglaman** [comment]
thanks!!

**Pull request merged**